tests/unit/App4.spec.ts

```diff
@@ -0,0 +1,463 @@
+            appointmentDate: '2018-01-01'
+          },
+          {
+            roleType: 'Director',
+            appointmentDate: '2018-01-01'
+          }
+        ]
+      },
+      {
+        officer: {
+          id: 2,
+          firstName: '',
+          lastName: '',
+          middleName: '',
+          organizationName: 'XyzInc.',
+          partyType: 'organization'
+        },
+        mailingAddress: {
+          streetAddress: 'mailing_address-addresslineone',
+          streetAddressAdditional: '',
+          addressCity: 'mailing_addresscity',
+          addressCountry: 'CA',
+          postalCode: 'H0H0H0',
+          addressRegion: 'BC'
+        },
+        appointmentDate: '2018-01-01',
+        roles: [
+          {
+            roleType: 'Incorporator',
+            appointmentDate: '2018-01-01'
+          }
+        ]
+      }
+    ],
+    shareStructure: {
+      shareClasses: [
+        {
+          id: 1,
+          name: 'ShareClass1',
+          priority: 1,
+          hasMaximumShares: true,
+          maxNumberOfShares: 100,
+          hasParValue: true,
+          parValue: 10,
+          currency: 'CAD',
+          hasRightsOrRestrictions: false,
+          series: [
+            {
+              id: 1,
+              name: 'ShareSeries1',
+              priority: 1,
+              hasMaximumShares: true,
+              maxNumberOfShares: 50,
+              hasRightsOrRestrictions: false
+            },
+            {
+              id: 2,
+              name: 'ShareSeries2',
+              priority: 2,
+              hasMaximumShares: true,
+              maxNumberOfShares: 100,
+              hasRightsOrRestrictions: false
+            }
+          ]
+        },
+        {
+          id: 2,
+          name: 'ShareClass2',
+          priority: 1,
+          hasMaximumShares: false,
+          maxNumberOfShares: null,
+          hasParValue: false,
+          parValue: null,
+          currency: null,
+          hasRightsOrRestrictions: true,
+          series: [
+
+          ]
+        }
+      ]
+    },
+    incorporationAgreement: {
+      agreementType: 'sample'
+    }
+  }
+}
+
+// sample NR data
+const nrData = {
+  applicants: {
+    addrLine1: 'address line 1',
+    addrLine2: 'address line 2',
+    addrLine3: 'address line 3',
+    city: 'Victoria',
+    countryTypeCd: 'CA',
+    emailAddress: 'tester@test.com',
+    firstName: 'John',
+    lastName: 'Doe',
+    middleName: 'Joe',
+    phoneNumber: '250-111-2222',
+    postalCd: 'V1V 1A2',
+    stateProvinceCd: 'BC'
+  },
+  consentFlag: 'R',
+  corpNum: null,
+  legalType: 'BEN',
+  expirationDate: 'Thu, 31 Dec 2099 23:59:59 GMT',
+  names: [
+    {
+      name: 'ABC 1234',
+      state: 'APPROVED'
+    },
+    {
+      name: 'CDE 1234',
+      state: 'NE'
+    }
+  ],
+  nrNum: 'NR 1234567',
+  request_action_cd: 'NEW',
+  state: 'APPROVED'
+}
+
+// *** TODO: implement more tests!
+describe('Restoration - Restore Business Name for a BEN', () => {
```

Comment:
Here's the new test suite for Restorations.

Additionally, we should have test suites for any new components. Those components can then be stubbed in higher-level tests (like the ones in THIS file).

Reply:
This describe block is now in App.spec.ts but same idea.